cluster-controller/src/main/java/io/strimzi/controller/cluster/operations/resource/AbstractOperations.java

```diff
@@ -165,4 +165,37 @@ public T get(String namespace, String name) {
         return operation().inNamespace(namespace).withLabels(labels).list().getItems();
     }
 
+    /**
+     * Waits until resource is in the Ready state
+     *
+     * @param namespace The namespace
+     * @param name The resource name
+     * @param timeout Timeout in {@code timeUnit}
+     * @param timeUnit Time unit
+     */
+    public Future<Void> waitUntilReady(String namespace, String name, long timeout, TimeUnit timeUnit) {
+        Future<Void> fut = Future.future();
+        vertx.createSharedWorkerExecutor("kubernetes-ops-pool").executeBlocking(
+                future -> {
+                    try {
+                        if (operation().inNamespace(namespace).withName(name).get() != null) {
+                            log.info("Waiting for {} resource {} in namespace {} to get ready", resourceKind, name, namespace);
+                            operation().inNamespace(namespace).withName(name).waitUntilReady(timeout, timeUnit);
```

Comment:
This will block, thus occupying a thread from the shared pool (you know this because you increased the size of the pool), but did you consider setting a vertx peroidic timer and checking `isReady()`, and completing the future when it returns true. That would result in better thread utilization.

Reply:
Might be something to look at ... the difference is that `waitUntilReady` should now if the pod is for example recreated. Periodic Timer might not. But that probably doesn't matter too much to us, so I can give it a try.

Reply:
@tombentley Ufff ... was harder than expected. `waitUntilReady` behaves consistently across all resources. `isReady` is supported only somewhere. But I guess at scale this should be worth it. Please have a look and review it - as our Java guru maybe you will have more elegant way how to handle the timer.